quickwit-search/src/leaf.rs

```diff
@@ -121,23 +125,83 @@ pub(crate) async fn open_index(
 ///
 /// The downloaded data depends on the query (which term's posting list is required,
 /// are position required too), and the collector.
+///
+/// * `query` - query is used to extract the terms and the fields which will be loaded from the
+/// inverted_index.
+///
+/// * `term_dict_field_names` - A list of fields, where the whole dictionary needs to be loaded.
+/// This is e.g. required for term aggregation, since we don't know in advance which terms are going
+/// to be hit.
 #[instrument(skip(searcher, query, fast_field_names))]
 pub(crate) async fn warmup(
     searcher: &Searcher,
     query: &dyn Query,
     fast_field_names: &HashSet<String>,
+    term_dict_field_names: &HashSet<String>,
 ) -> anyhow::Result<()> {
     let warm_up_terms_future =
         warm_up_terms(searcher, query).instrument(debug_span!("warm_up_terms"));
+    let warm_up_term_dict_future = warm_up_term_dict_fields(searcher, term_dict_field_names)
+        .instrument(debug_span!("warm_up_term_dicts"));
     let warm_up_fastfields_future = warm_up_fastfields(searcher, fast_field_names)
         .instrument(debug_span!("warm_up_fastfields"));
-    let (warm_up_terms_res, warm_up_fastfields_res) =
-        tokio::join!(warm_up_terms_future, warm_up_fastfields_future);
+    let (warm_up_terms_res, warm_up_fastfields_res, warm_up_term_dict_res) = tokio::join!(
+        warm_up_terms_future,
+        warm_up_fastfields_future,
+        warm_up_term_dict_future
+    );
     warm_up_terms_res?;
     warm_up_fastfields_res?;
+    warm_up_term_dict_res?;
+    Ok(())
+}
+
+async fn warm_up_term_dict_fields(
+    searcher: &Searcher,
+    term_dict_field_names: &HashSet<String>,
+) -> anyhow::Result<()> {
+    let mut term_dict_fields = Vec::new();
+    for term_dict_field_name in term_dict_field_names.iter() {
+        let term_dict_field = searcher
+            .schema()
+            .get_field(term_dict_field_name)
+            .with_context(|| {
+                format!(
+                    "Couldn't get field named {:?} from schema.",
+                    term_dict_field_name
+                )
+            })?;
+
+        term_dict_fields.push(term_dict_field);
+    }
+
+    let mut warm_up_futures = Vec::new();
+    for field in term_dict_fields {
+        for segment_reader in searcher.segment_readers() {
+            let inverted_index = segment_reader.inverted_index(field)?.clone();
+            warm_up_futures.push(async move {
+                let dict = inverted_index.terms();
+                dict.warm_up_dictionary().await
+            });
+        }
+    }
+    try_join_all(warm_up_futures).await?;
     Ok(())
 }
 
+pub fn get_fastfield_cardinality(field_type: &FieldType) -> Option<Cardinality> {
```

Comment:
Thanks for thinking about the name. (calling it fastfield_cardinality.)
The notion of cardinality is indeed ambiguous.

A single valued bytes field in quickwit has a multivalued fast field cardinality.

Maybe adding a comment would be a good addition. 
(and make the function private)